addons/binding/org.openhab.binding.sensebox/src/main/java/org/openhab/binding/sensebox/internal/handler/SenseBoxHandler.java

```diff
@@ -243,7 +251,49 @@ private State decimalFromSensor(SenseBoxSensor data) {
         if (data != null) {
             if (data.getLastMeasurement() != null) {
                 if (StringUtils.isNotEmpty(data.getLastMeasurement().getValue())) {
-                    result = new DecimalType(data.getLastMeasurement().getValue());
+                    logger.debug("About to determine quantity for {} / {}", data.getLastMeasurement().getValue(), data.getUnit());
+                    BigDecimal bd = new BigDecimal(data.getLastMeasurement().getValue());
+
+                    try {
+                        switch (data.getUnit()) {
+                            case "%":
+                                result = new QuantityType<>(bd, SmartHomeUnits.ONE);
+                                break;
+                            case "°C":
+                                result = new QuantityType<>(bd, SIUnits.CELSIUS);
+                                break;
+                            case "Pa":
+                                result = new QuantityType<>(bd, SIUnits.PASCAL);
+                                break;
+                            case "hPa":
+                                if (BigDecimal.valueOf(10000l).compareTo(bd) < 0) {
+                                    // Some stations report measurements in Pascal, but send 'hPa' as units...
+                                    bd = bd.divide(ONEHUNDRED);
+                                }
+                                result = new QuantityType<>(bd, HECTO(SIUnits.PASCAL));
+                                break;
+                            case "lx":
+                                result = new QuantityType<>(bd, SmartHomeUnits.LUX);
+                                break;
+                            case "µg/m³":
+                                // result = new QuantityType<>(bd, SmartHomeUnits.MICROGRAM_PER_CUBICMETRE);
+                                result = new QuantityType<>(bd, SmartHomeUnits.KILOGRAM_PER_CUBICMETRE);
+                                break;
+                            case "μW/cm²":
+                                // result = new QuantityType<>(bd, SmartHomeUnits.MICROWATT_PER_SQUARE_CENTIMETRE);
+                                result = new QuantityType<>(bd, SmartHomeUnits.IRRADIANCE);
+                                break;
+                            default:
+                                logger.info("    Could not determine unit for '{}', using default", data.getUnit());
+                                result = new QuantityType<>(bd, SmartHomeUnits.ONE);
+                        }
+                    } catch (java.lang.IllegalStateException ise) {
+                        // java.lang.IllegalStateException: Error invoking #valueOf(String) on class 'org.eclipse.smarthome.core.library.types.QuantityType' with value '0.00 μg/m³'.
```

Comment:
Can you also add an explanation to the comment in what scenarios the exception might occur?

Reply:
Even though I am running my binding in OH Build 1490, and the two PRs for the new units should be available, I am thrown an Exception from the underlying library that my unit description could not be parsed. The java unit tests in the smarthome core run successfully though.

So, until I discover the root cause of the parsing errors, I'd like to have some uom support for those two measurements and in the meantime continue debugging the core.

Reply:
Do you have a full stacktrace of those exceptions and example `BigDecimal` values?

Reply:
> Do you have a full stacktrace of those exceptions and example `BigDecimal` values?

Here ist the full stacktrace from openhab.log, and the corresponding BigDecimal value is 2.60:

```
2019-01-03 20:45:33.125 [ERROR] [me.core.internal.events.EventHandler] - Creation of ESH-Event failed, because one of the registered event factories has thrown an exception: Error invoking #valueOf(String) on class 'org.eclipse.smarthome.core.library.types.QuantityType' with value '2.60 μg/m³'.
java.lang.IllegalStateException: Error invoking #valueOf(String) on class 'org.eclipse.smarthome.core.library.types.QuantityType' with value '2.60 μg/m³'.
        at org.eclipse.smarthome.core.items.events.ItemEventFactory.parseSimpleClassName(ItemEventFactory.java:190) ~[?:?]
        at org.eclipse.smarthome.core.items.events.ItemEventFactory.parseType(ItemEventFactory.java:158) ~[?:?]
        at org.eclipse.smarthome.core.items.events.ItemEventFactory.getState(ItemEventFactory.java:136) ~[?:?]
        at org.eclipse.smarthome.core.items.events.ItemEventFactory.createStateEvent(ItemEventFactory.java:116) ~[?:?]
        at org.eclipse.smarthome.core.items.events.ItemEventFactory.createEventByType(ItemEventFactory.java:80) ~[?:?]
        at org.eclipse.smarthome.core.events.AbstractEventFactory.createEvent(AbstractEventFactory.java:51) ~[?:?]
        at org.eclipse.smarthome.core.internal.events.EventHandler.createESHEvent(EventHandler.java:121) ~[?:?]
        at org.eclipse.smarthome.core.internal.events.EventHandler.handleEvent(EventHandler.java:95) ~[?:?]
        at org.eclipse.smarthome.core.internal.events.EventHandler.handleEvent(EventHandler.java:72) ~[?:?]
        at org.eclipse.smarthome.core.internal.events.ThreadedEventHandler.lambda$0(ThreadedEventHandler.java:67) ~[?:?]
        at java.lang.Thread.run(Thread.java:748) [?:?]
Caused by: java.lang.reflect.InvocationTargetException
        at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method) ~[?:?]
        at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62) ~[?:?]
        at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43) ~[?:?]
        at java.lang.reflect.Method.invoke(Method.java:498) ~[?:?]
        at org.eclipse.smarthome.core.items.events.ItemEventFactory.parseSimpleClassName(ItemEventFactory.java:181) ~[?:?]
        ... 10 more
Caused by: java.lang.IllegalArgumentException: μg not recognized (in 2.60 μg/m³ at index 5)
        at tec.uom.se.quantity.Quantities.getQuantity(Quantities.java:80) ~[?:?]
        at org.eclipse.smarthome.core.library.types.QuantityType.<init>(QuantityType.java:96) ~[?:?]
        at org.eclipse.smarthome.core.library.types.QuantityType.valueOf(QuantityType.java:139) ~[?:?]
        at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method) ~[?:?]
        at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62) ~[?:?]
        at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43) ~[?:?]
        at java.lang.reflect.Method.invoke(Method.java:498) ~[?:?]
        at org.eclipse.smarthome.core.items.events.ItemEventFactory.parseSimpleClassName(ItemEventFactory.java:181) ~[?:?]
        ... 10 more

```

Reply:
Thanks for the stack trace. I made a few additional unit tests and could reproduce/fix it. See https://github.com/eclipse/smarthome/pull/6818.

Reply:
Thanks for updating the PR with the correct Micro symbol. 

Did you already already test if the new units properly work with my pending ESH PR?

If it works, the try/catch IllegalStateException can be removed as well. :slightly_smiling_face:
That's the only remaining comment I have for this PR!

Reply:
> Did you already already test if the new units properly work with my pending ESH PR?

Not yet, it is slightly harder for me to test things that depend on the smarthome core. I have to have a full rebuild of the core with your fix, and then start my own part again. I was really hoping that your PR can be merged, a new smarthome release built, and then I can retest my part on top of it.

But if that takes too long or is too risk-prone, I might have a go at the smarthome core. I can't promise that it would be before the weekend, though...

> If it works, the try/catch IllegalStateException can be removed as well. 🙂
> That's the only remaining comment I have for this PR!

Reply:
> Did you already already test if the new units properly work with my pending ESH PR?

Good news. After building the core with https://github.com/eclipse/smarthome/pull/6818 and my PR, everything works fine and displays the correct units:

![sensebox-uom-working](https://user-images.githubusercontent.com/352641/50903463-2c850400-141e-11e9-86b3-9d5c96776cf7.png)

So, from my side, we should be good to go to merge the smarthome part. In this PR, I'll remove the try/catch later today

Reply:
> In this PR, I'll remove the try/catch later today

And the try/catch is removed...

Reply:
Nice! LGTM  now. Let's wait for the other PR to be merged now. :smiley:  :hourglass_flowing_sand: